src/algebra/opposites.lean

```diff
@@ -284,3 +285,10 @@ def ring_hom.to_opposite {R S : Type*} [semiring R] [semiring S] (f : R →+* S)
 
 @[simp] lemma ring_hom.coe_to_opposite {R S : Type*} [semiring R] [semiring S] (f : R →+* S)
   (hf : ∀ x y, commute (f x) (f y)) : ⇑(f.to_opposite hf) = op ∘ f := rfl
+
+/-- Used mainly to define the natural topology of `units α`. -/
+def embed_product (α : Type*) [monoid α] : units α →* α × αᵒᵖ :=
```

Comment:
I'd be inclined to put this in `algebra.group.prod`; I think of `algebra.opposites` rather as a bare-basics helper file (for example, opposites material about powers is in `algebra.group_power`, opposites material about the geometric series is in `algebra.geom_sum`).